src/test/java/com/mathworks/ci/RunMatlabBuildStepTest.java

```diff
@@ -0,0 +1,158 @@
+package com.mathworks.ci;
+
+/**
+ * Copyright 2022 The MathWorks, Inc.
+ *  
+ */
+
+import java.io.IOException;
+
+import hudson.model.Result;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+import hudson.FilePath;
+import hudson.slaves.DumbSlave;
+
+public class RunMatlabBuildStepTest {
+
+
+    private WorkflowJob project;
+
+
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Before
+    public void testSetup() throws IOException {
+        this.project = j.createProject(WorkflowJob.class);
+    }
+
+
+    /*
+     * Verify when MATLAB is not on system path.
+     */
+    @Test
+    public void verifyMATLABPathNotSet() throws Exception {
+        project.setDefinition(
+                new CpsFlowDefinition("node { runMATLABBuild() }", true));
+        WorkflowRun build = project.scheduleBuild2(0).get();
+        j.assertLogContains("MATLAB_ROOT", build);
+    }
+
+    /*
+     * Verify MATLAB gets invoked from workspace.
+     */
+    @Test
+    public void verifyMATLABstartsInWorkspace() throws Exception {
+        DumbSlave s = j.createOnlineSlave();
+        project.setDefinition(
+                new CpsFlowDefinition("node('!master') { runMATLABBuild() }", true));
+
+        FilePath workspace = s.getWorkspaceFor(project);
+        String workspaceName = workspace.getName();
+        WorkflowRun build = project.scheduleBuild2(0).get();
+
+        j.assertLogContains(workspaceName, build);
+    }
+
+    /*
+     * Verify MATLAB is invoked when valid MATLAB is on system path.
+     */
+    @Test
+    public void verifyMATLABPathSet() throws Exception {
+        project.setDefinition(
+                new CpsFlowDefinition("node { testMATLABBuild() }", true));
+        WorkflowRun build = project.scheduleBuild2(0).get();
+        j.assertLogContains("tester_started", build);
+    }
+
+    /*
+     * Verify Pipeline script runs on Jenkins node with valid MATLAB
+     */
+    @Test
+    public void verifyPipelineOnSlave() throws Exception {
+        DumbSlave s = j.createOnlineSlave();
+        project.setDefinition(new CpsFlowDefinition(
+                "node('!master') { testMATLABBuild() }", true));
+
+        s.getWorkspaceFor(project);
+        WorkflowRun build = project.scheduleBuild2(0).get();
+
+        j.assertBuildStatusSuccess(build);
+    }
+
+    /*
+     * Verify appropriate task is invoked as in pipeline script
+     */
+    @Test
+    public void verifyTasksSameAsScript() throws Exception {
+        project.setDefinition(
+                new CpsFlowDefinition("node { runMATLABBuild(tasks: 'compile') }", true));
```

Comment:
A lot of these tests (and the builder tests) feel more like integration tests, and several of them take a while to run. Maybe some abstractions can make the code more directly testable

Reply:
alternately as i commented earlier we could just restrict test related to RunMatlabBuildstep class.